### PR TITLE
Feature/unr 4255 test expected log

### PR DESCRIPTION
**Pull request by nafonso**
#### Description
Added AddExpectedError API that allows you to flag warnings / errors as expected to appear, hence not making tests fail when they do.

#### Tests
Include SpatialTestNetOwnership fix to be able to always pass with green instead of "passing with warnings"

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp [hunk 1]
```diff
@@ -367,6 +369,31 @@ void ASpatialFunctionalTest::FinishTest(EFunctionalTestResult TestResult, const
...
+{
+	if (HasAuthority())
+	{
+		FAutomationTestBase* CurrentTest = FAutomationTestFramework::Get().GetCurrentTest();
+		check(CurrentTest != nullptr);
+		for (const auto& ExpectedError : ExpectedErrors)
+		{
+			CurrentTest->AddExpectedError(
+				ExpectedError.ErrorPattern,
+				ExpectedError.bAllowPartialMatch ? EAutomationExpectedErrorFlags::Contains : EAutomationExpectedErrorFlags::Exact,
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:391]: I think we could expose `EAutomationExpectedErrorFlags` directly, but minor thing.

Comment (nafonso) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:391]: I didn't specifically to not change the engine, if you prefer I can, but need to change it to enum class, etc.

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:391]: Hm? Why does this require changing the engine? I thought this was exposed for use in the AutomationTests. Maybe will need to catch up "offline", but I think there should exist a configuration that allows us to use it :D.

**Review by mironec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp [hunk 2]
```diff
@@ -367,6 +368,12 @@ void ASpatialFunctionalTest::FinishTest(EFunctionalTestResult TestResult, const
 	}
 }
 
+void ASpatialFunctionalTest::AddExpectedLogError(const FString& ExpectedPatternString, int32 Occurrences /*= 1*/,
+												 bool ExactMatch /*= false*/)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:372]: bExactMatch

**Comment by m-samiec** [comment]
Is there a reason on adding this over just calling `UAutomationBlueprintFunctionLibrary::AddExpectedLogError` directly?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp (hunk 2 above)
Comment (nafonso) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:372]: It was a copy of the API defined by the blueprint library, but I'll change it for ours.

**Comment by nafonso** [comment]
> Is there a reason on adding this over just calling `UAutomationBlueprintFunctionLibrary::AddExpectedLogError` directly?

The same reason that we thought that there was no functionality for it, because there's no API in the FunctionalTests. I found out about it because it shows up in the BP when you search.

**Comment by m-samiec** [comment]
> The same reason that we thought that there was no functionality for it, because there's no API in the FunctionalTests. I found out about it because it shows up in the BP when you search.

Okay, expose it for blueprint access then?

**Comment by nafonso** [comment]
> > The same reason that we thought that there was no functionality for it, because there's no API in the FunctionalTests. I found out about it because it shows up in the BP when you search.
> 
> Okay, expose it for blueprint access then?

If I do, then they'll have the same function exposed twice, so I opted to let the blueprint library handle it for BPs and this one doing it for cpp.

**Review by m-samiec — APPROVED** [review]

**Pull request merged**